lib/Mojo/WebSocketProxy/Backend/ConsumerGroups.pm

```diff
@@ -307,10 +312,19 @@ sub wait_for_messages {
 sub _on_message {
     my ($self, $redis, $raw_message) = @_;
 
-    my $message = eval { decode_json_utf8($raw_message) };
+    my $message = {};
+
+    try{
+        $message = decode_json_utf8($raw_message);
+    }catch {
+        my $err = $@;
 
-    unless(ref $message eq 'HASH' && $message->{message_id}) {
-        $log->errorf('Fail to proccess response: %s', $raw_message);
+        $log->errorf('An error occurred while decoding published response by Consumer (Transport::Redis):', $err);
+        return;
+    }
+
+    if(!$message->{message_id}) {
```

Comment:
this data is from out side you have no guarantee that it'll be a hash. in case if it's a arrayref, you'll get exception here.

Reply:
How might it be `arrayref` since we always encode `hashrefs` before publish in the consumer part?

Reply:
@mehdi-fs  This repository and module is open-source and independent of any business logic or company repos like `binary-websocket-api`. So it doesn't matter how `we` are using it; it should be developed and reviewed with putting our awareness of `binary-websocket-api` code in brackets.

Reply:
@mat-fs I didn't know that, it makes sense, I will fix it

Reply:
Yes. All repos under `binary-com` github company name are open source and public access.